File: lista3/java/figury.java
public class figury {
    public static void main(String[] args){
        if(args.length==0){
            System.out.println("Brak argumentow!");
            System.exit(0);
        }
        String typyFigur = args[0];
        Figura [] figury = new Figura[typyFigur.length()];
        int wskaznik = 1;
        int iloscArgumentow = args.length;
        for (int i = 0; i < typyFigur.length(); i++){
            char znak = typyFigur.charAt(i);
            try {
                if (znak == 'o') { //koło
                    if (wskaznik + 1 > iloscArgumentow) {
                        System.out.println("Brakujace parametry!");
                        System.exit(0);
                    }
                    int promien = Integer.valueOf(args[wskaznik]);
                    wskaznik++;
                    figury[i] = new Kolo(promien);

                } else if (znak == 'c') { //czworokat
                    if (wskaznik + 5 > iloscArgumentow) {
                        System.out.println("Brakujace parametry!");
                        System.exit(0);
                    }
                    int bok1 = Integer.valueOf(args[wskaznik]);
                    wskaznik++;
                    int bok2 = Integer.valueOf(args[wskaznik]);
                    wskaznik++;
                    int bok3 = Integer.valueOf(args[wskaznik]);
                    wskaznik++;
                    int bok4 = Integer.valueOf(args[wskaznik]);
                    wskaznik++;
                    int kat = Integer.valueOf(args[wskaznik]);
                    wskaznik++;

                    if((bok1==bok2 && bok3==bok4)||(bok1==bok3 && bok2==bok4)||(bok1==bok4 && bok2==bok3)){//jesli rownoleglobok
                        if(bok1==bok2&&bok2==bok3&&bok3==bok4){//jesli rowne boki
                            if(kat==90){ //tworzymy kwadrat
                                figury[i] = new Kwadrat(bok1);
                            }
                            else         //tworzymy romb
                                figury[i] = new Romb(bok1, kat);
                        }
                        else if(kat==90){ //tworzymy prostokat
                            int bokX, bokY;
                            bokX = bok1;
                            if(bokX != bok2)
                                bokY = bok2;
                            else if(bokX!=bok3)
                                bokY = bok3;
                            else
                                bokY = bok4;
                            figury[i] = new Prostokat(bokX, bokY);
                        }
                        else{ // rownoleglobok nie jest prostokatem, rombem ani kwadratem
                            throw new Exception("Figura jest rownoleglobokiem, ale nie jest prostokatem, rombem ani kwadratem");
                        }
                    }
                    else{ // rownoleglobok nie jest prostokatem, rombem ani kwadratem
                        throw new Exception("Figura nie jest prostokatem, rombem ani kwadratem");
                    }

                } else if (znak == 'p') { //pieciokat
                    if (wskaznik + 1 > iloscArgumentow) {
                        System.out.println("Brakujace parametry!");
                        System.exit(0);
                    }
                    int bok = Integer.valueOf(args[wskaznik]);
                    wskaznik++;

                    figury[i] = new Pieciokat(bok);

                } else if (znak == 's') { //szesciokat
                    if (wskaznik + 1 > iloscArgumentow) {
                        System.out.println("Brakujace parametry!");
                        System.exit(0);
                    }
                    int bok = Integer.valueOf(args[wskaznik]);
                    wskaznik++;

                    figury[i] = new Szesciokat(bok);

                }
            }
            catch(Exception e){
                System.out.println(e);
            }
        }
        for(Figura figura : figury){
            if(figura != null)
                System.out.printf("%s: pole: %.2f obwod: %.2f\n", figura.nazwa, figura.Pole(),figura.Obwod());
        }
    }
}

File: lista3/c++/figury.cpp
#include <iostream>
#include <math.h>
#include <cstdio>
#include <vector>
#include <locale>
#include <sstream>
#define PI 3.14159265
using namespace std;
class Figura
{
public:
    virtual float pole() = 0;
    virtual float obwod() = 0;
    string nazwa;
};
class Kolo : public Figura
{
public:

    int promien;
    Kolo(int r)
    {
        promien = r;
        nazwa = "kolo";
    }
    float pole()
    {
        return PI * promien*promien;
    }
    float obwod()
    {
        return PI * 2 * promien;
    }
};
class Czworokat : public Figura
{
public:
    float bok1 = 0, bok2 = 0, bok3 = 0, bok4 = 0;
    int kat = 0;
    Czworokat(int boka, int bokb, int bokc, int bokd, int k)
    {
        bok1 = boka;
        bok2 = bokb;
        bok3 = bokc;
        bok4 = bokd;
        kat = k;
    }
    float obwod()
    {
        return bok1 + bok2 + bok3 + bok4;
    }

};
class Kwadrat : public Czworokat
{
public:
    Kwadrat(int boka, int bokb, int bokc, int bokd, int k) : Czworokat(boka,bokb,bokc,bokd,k)
    {
        nazwa = "kwadrat";
    }
    float pole()
    {
        return bok1 * bok1;
    }


};
class Romb : public Czworokat
{
public:
    Romb(int boka, int bokb, int bokc, int bokd, int k) : Czworokat(boka, bokb, bokc, bokd, k)
    {
        nazwa = "romb";
    }
    float pole()
    {
        return bok1 * bok1*sin(kat*PI / 180);
    }

};
class Prostokat : public Czworokat
{
public:
    Prostokat(int boka, int bokb, int bokc, int bokd, int k) : Czworokat(boka, bokb, bokc, bokd, k)
    {
        nazwa = "prostokat";
    }
    float pole()
    {
        if (bok1 == bok2)
            return bok1 * bok3;
        else
            return bok1 * bok2;
    }

};
class Pieciokat : public Figura {
public:
    int bok;
    Pieciokat(int b)
    {
        bok = b;
        nazwa = "pieciokat";
    }
    float pole()
    {
        return (float) ( ((bok*bok)* sqrt(25+10*sqrt(5)))/4 );
    }
    float obwod()
    {
        return 5*bok;
    }



};
class Szesciokat : public Figura {
    public:
    int bok;
     Szesciokat(int b) {
        bok = b;
        nazwa = "Szesciokat";
    }
     float pole() {
        return (float)((3*bok*bok*sqrt(3))/2);
    }
     float obwod() {
        return 6*bok;
    }
};
int zamien(string a)
{
    int k = 0;
    for (int i = 0; i < a.size(); i++)
    {
        k *= 10;
        if (!isdigit(a[i]))
        {
            return -1;
        }
        k += (int)(a[i] - '0');
    }
    return k;
}
bool dodatnia(int a[5])
{
    for(int i = 0; i<5; i++)
    {
        if(a[i]<=0)return false;
    }
    return true;
}
int main(int max_pozycja, char *argv[])
{
    vector <Figura*> figury;
    string args = argv[1];
    int pozycja = 2;
    max_pozycja--;
    for (int i = 0; i < args.size(); i++)
    {

        char c = args[i];
        if (c == 'o')
        {
            if (pozycja > max_pozycja)
            {
                cout<<"Figura numer "<< i << " Koniec danych!\n";
            }
            else
            {
                Kolo *kolo = new Kolo(zamien(argv[pozycja]));
                if (kolo->promien <= 0)
                    cout<<"Figura numer "<< i << " - niepoprawnie podane dane\n";
                else
                    figury.push_back(kolo);
                pozycja++;
            }
        }
        if(c=='p')
        {
            if(pozycja > max_pozycja)
            {
                cout<<"Figura numer " << i << " Koniec danych!\n";
            }
            else
            {
                Pieciokat *pieciokat = new Pieciokat(zamien(argv[pozycja]));
                pozycja++;
                if(pieciokat->bok <= 0)
                    cout<<"Figura numer " << i << " - niepoprawnie podane dane\n";
                else
                    figury.push_back(pieciokat);
            }
        }
        if(c=='s')
        {
            if(pozycja > max_pozycja)
            {
                cout<<"Figura numer " << i << " Koniec danych!\n";
            }
            else
            {
                Szesciokat *szesciokat = new Szesciokat(zamien(argv[pozycja]));
                pozycja++;
                if(szesciokat->bok <= 0)
                    cout<<"Figura numer " << i << " - niepoprawnie podane dane\n";
                else
                    figury.push_back(szesciokat);
            }
        }
           if(c=='c') {
                if(pozycja + 4 > max_pozycja) {
                    cout<<"Figura numer " << i << " Koniec danych!\n";
                }
                else {


                    int b[5];
                    for (int k = 0; k < 5; k++) {
                        b[k] = zamien(argv[pozycja]);
                        pozycja++;
                    }
                    if (!dodatnia(b)) {
                        cout<<"Figura numer " << i << " - niepoprawnie podane dane\n";
                    } else {
                        if (b[0] == b[1] && b[0] == b[2] && b[0] == b[3]) {
                            if (b[4] == 90) {
                                Kwadrat *kwadrat = new Kwadrat(b[0], b[1], b[2], b[3], b[4]);
                                figury.push_back(kwadrat);
                            } else if (b[4] < 180 && b[4] > 0) {
                                Romb *romb = new Romb(b[0], b[1], b[2], b[3], b[4]);
                                figury.push_back(romb);
                            }
                        } else if (((b[0] == b[1] && b[2] == b[3]) || (b[0] == b[2] && b[3] == b[1]) || (b[0] == b[3] && b[1] == b[2])) && b[4] == 90) {
                            Prostokat *prostokat = new Prostokat(b[0], b[1], b[2], b[3], b[4]);
                            figury.push_back(prostokat);
                        } else cout<<"Figura numer " << i << " - niepoprawnie podane dane\n";
                    }


                }
            }
    }
    for(int i =0; i<figury.size(); i++)
    {
        cout << figury[i]->nazwa << " obwod: " << figury[i]->obwod() << " pole: " << figury[i]->pole()<<'\n';
    }
    return 0;
}

